app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/WifiDirectManager.java

```diff
@@ -0,0 +1,466 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.IntentFilter;
+import android.net.Uri;
+import android.net.wifi.WpsInfo;
+import android.net.wifi.p2p.WifiP2pConfig;
+import android.net.wifi.p2p.WifiP2pDevice;
+import android.net.wifi.p2p.WifiP2pDeviceList;
+import android.net.wifi.p2p.WifiP2pInfo;
+import android.net.wifi.p2p.WifiP2pManager;
+import android.os.Build;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function0;
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.AlertDialogShower;
+import org.kiwix.kiwixmobile.utils.KiwixDialog;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import static android.os.Looper.getMainLooper;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.LocalFileTransferActivity.showToast;
+
+
+/**
+ * Manager for the Wifi-P2p API, used in the local file transfer module
+ * */
+public class WifiDirectManager implements WifiP2pManager.ChannelListener, WifiP2pManager.PeerListListener, WifiP2pManager.ConnectionInfoListener,
+    KiwixWifiP2pBroadcastReceiver.P2pEventListener {
+
+  private static final String TAG = "WifiDirectManager";
+  public static int FILE_TRANSFER_PORT = 8008;
+
+  private @NonNull LocalFileTransferActivity activity;
+
+  private SharedPreferenceUtil sharedPreferenceUtil;
+  private AlertDialogShower alertDialogShower;
+
+  /* Variables related to the WiFi P2P API */
+  private boolean wifiP2pEnabled = false; // Whether WiFi has been enabled or not
+  private boolean retryChannel = false;   // Whether channel has retried connecting previously
+
+  private WifiP2pManager manager;         // Overall manager of Wifi p2p connections for the module
+  private WifiP2pManager.Channel channel;
+  // Connects the module to device's underlying Wifi p2p framework
+
+  private BroadcastReceiver receiver = null; // For receiving the broadcasts given by above filter
+
+  private WifiP2pDevice userDevice;   // Represents the device on which the app is running
```

Comment:
you don't need this variable

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/227b5228f8dc09c7ed8351397c3854759516d9a3#diff-7e01f6f425daaca67175b74c839c6af2L61)
I kept it around expecting something might pop up, but I guess less code is always the best code :+1: